agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/JsonConfigPolling.java

```diff
@@ -42,19 +45,22 @@
 
     private final Path path;
     private volatile long lastModifiedTime;
+    private volatile double lastReadSamplingPercentage;
     private static final Logger logger = LoggerFactory.getLogger(JsonConfigPolling.class);
 
-    private JsonConfigPolling(Path path, long lastModifiedTime) {
+    private JsonConfigPolling(Path path, long lastModifiedTime, double lastReadSamplingPercentage) {
         this.path = path;
         this.lastModifiedTime = lastModifiedTime;
+        this.lastReadSamplingPercentage = lastReadSamplingPercentage;
     }
 
-    public static void pollJsonConfigEveryMinute(Path path, long lastModifiedTime) {
+    public static void pollJsonConfigEveryMinute(Path path, long lastModifiedTime, double lastReadSamplingPercentage) {
```

Comment:
is it necessary to pass in this lastReadSamplingPercentage?  We have Global.getSamplingPercentage.  Is it because it was rounded, which lost precision?